torch/csrc/xpu/Event.cpp

```diff
@@ -0,0 +1,173 @@
+#include <pybind11/pybind11.h>
+#include <torch/csrc/Device.h>
+#include <torch/csrc/THP.h>
+#include <torch/csrc/utils/pybind.h>
+#include <torch/csrc/utils/pycfunction_helpers.h>
+#include <torch/csrc/utils/python_arg_parser.h>
+#include <torch/csrc/xpu/Event.h>
+#include <torch/csrc/xpu/Module.h>
+#include <torch/csrc/xpu/Stream.h>
+
+#include <structmember.h>
+
+PyObject* THXPEventClass = nullptr;
+
+static PyObject* THXPEvent_pynew(
+    PyTypeObject* type,
+    PyObject* args,
+    PyObject* kwargs) {
+  HANDLE_TH_ERRORS
+  unsigned char enable_timing = 0;
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+  constexpr const char* kwlist[] = {"enable_timing", nullptr};
+  if (!PyArg_ParseTupleAndKeywords(
+          args, kwargs, "|b", const_cast<char**>(kwlist), &enable_timing)) {
+    return nullptr;
+  }
+
+  THPObjectPtr ptr(type->tp_alloc(type, 0));
+  if (!ptr) {
+    return nullptr;
+  }
+
+  THXPEvent* self = (THXPEvent*)ptr.get();
+
+  new (&self->xpu_event) at::xpu::XPUEvent(enable_timing);
+
+  return (PyObject*)ptr.release();
+  END_HANDLE_TH_ERRORS
+}
+
+static void THXPEvent_dealloc(THXPEvent* self) {
+  self->xpu_event.~XPUEvent();
```

Comment:
FYI this now has a gil release in the cuda side https://github.com/pytorch/pytorch/pull/119393
If the event destruction might trigger non-trivial code and block, you should do the same

Reply:
Thanks for your information. If `cudaEventDestroy` is called while the associated event hasn't been completed, the CUDA runtime will wait for those operations to complete asynchronously before destroying the CUDA event. I think SYCL runtime might also encounter similar issues, although I cannot confirm it one hundred percent.

Reply:
Add gil release code.